modules/ui/src/main/scala/lucuma/ui/forms/FormInputEV.scala

```diff
@@ -53,92 +55,135 @@ final case class FormInputEV[EV[_], A](
   transparent:     js.UndefOr[Boolean] = js.undefined,
   width:           js.UndefOr[SemanticWidth] = js.undefined,
   value:           EV[A],
-  format:          InputFormat[A] = InputFormat.id,
+  validate:        InputValidate[A] = InputValidate.id,
   modifiers:       Seq[TagMod] = Seq.empty,
-  onChange:        FormInputEV.ChangeCallback[A] =
-    (_: A) => Callback.empty, // callback for parents of this component
-  onBlur:          FormInputEV.ChangeCallback[A] = (_: A) => Callback.empty
+  onChange:        FormInputEV.ChangeCallback[A] = (_: A) => Callback.empty, // for extra actions
+  onValidChange:   FormInputEV.ChangeCallback[Boolean] = _ => Callback.empty,
```

Comment:
So would this be called only when we have a valid `A` and on what event?

Reply:
It seems we could reuse the same callback with an auto save timer at the input level

Reply:
I believe this is called with every keystroke based on whether the current input is valid or not.

Reply:
Ah `onValidChange` is called on every keystroke with the String value?
and `onChange` with every keystroke if the value can be parsed?
Is it so?

Reply:
The implemented behavior is as follows, sorry for not explaining better in the PR description.

Partial validation during typing:
* `onValidChange` is a `Boolean => Callback` and is called on every keystroke, informing the parent whether the current input is valid or not. This allows, for example, enabling or disabling buttons in the parent that depend upon the input being valid.
* `onChanged` is a `A => Callback` and is also called on every keystroke, but only if validation is successful and an `A` value can be obtained from the input after the keystroke. This probably will be used rarely. I think that behavior that relies on the value should be triggered when the value is set in the model (see `onBlur` below). Maybe we can remove this callback altogether, or switch it to a `String => Callback`.
* During typing no errors are displayed, nor is the model value changed.

Full validation `onBlur`:
* Upon blur, errors are shown. Otherwise, if vaidation passes, then the model value is changed.
* Also, `onBlur` callback is called in case the parent wants to add additional behavior. This is currently a `A => Callback` but maybe it makes more sense to change it to a `Validated[A] => Callback`.

Other notes:
* `InputValidate` is an optic that combines `Format` and validation. Investigating how and if it composes and its laws is still pending.

Reply:
Thanks for the explanation. I like the `onValidChange` a lot, it provides a good place to interact with the parent.
I also agree `onChanged` may not be necessary